--- 2022.01.24.1903/script/start_main_service.sh
+++ 2022.06.08.1109/script/start_main_service.sh
@@ -9,6 +9,7 @@
 #!/bin/bash
 #open kernel log timestamp
 echo y > /sys/module/printk/parameters/time
+echo 4  > /proc/sys/kernel/printk
 #set date
 date -s 2020-01-02
 
@@ -28,6 +29,7 @@
 fi
 
 if [ -e /usr/sbin/riddleBoxCfg ]; then 
+	test -e /etc/riddle.conf || (echo "create riddle.conf from default config file";cp /etc/riddle_default.conf /etc/riddle.conf)
 	test -e /etc/riddle.conf || (echo "create riddle.conf from old config file";riddleBoxCfg --readOld)
 fi
 	
@@ -65,8 +67,10 @@
 		./fakeCarLifeDevice >> /dev/null 2>&1 &
 	fi
 elif [ -e /usr/sbin/ARMadb-driver ]; then
+	sleep `riddleBoxCfg -g BoxConfig_DelayStart`
 	insmod /tmp/storage_common.ko
 	insmod /tmp/g_android_accessory.ko
+	/script/start_accessory.sh &
 	insmod /tmp/cdc_ncm.ko
 	if [ -e /etc/log_com ]; then
 		ARMadb-driver 2>&1 | tee -a /tmp/userspace.log &
@@ -79,6 +83,8 @@
 	else
 		ARMadb-driver >> /dev/null 2>&1 &
 	fi
+	#A15 not need activation, remove old version boa web 
+	test -e /etc/boa_old && rm -rf /etc/boa_old
 fi
 
 
